### PR TITLE
Refactor/auction module

### PR DESCRIPTION
**Pull request by danimhr**
This PR aims to add auction module to per server.

### CODE REVIEW

File: auction-server/src/auction/repository/add_recent_priotization_fee.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            .write()
+            .await;
+        write_guard.push_back(PrioritizationFeeSample {
+            fee,
+            sample_time: OffsetDateTime::now_utc(),
+        });
+        if write_guard.len() > RECENT_FEES_SLOT_WINDOW {
+            write_guard.pop_front();
+        }
+        tracing::info!("Recent prioritization fees: {:?}", write_guard);
```
Comment (m30m) [inline auction-server/src/auction/repository/add_recent_priotization_fee.rs:28]: this is also updated on the main branch now. We are just logging the one we received, not all.

File: auction-server/src/auction/service/auctionable.rs [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+        result,
+        time::Duration,
+    },
+    time::OffsetDateTime,
+    uuid::Uuid,
+};
+
+/// The trait for handling the auction for the service.
+#[async_trait]
+pub trait Auctionable<T: ChainTrait> {
```
Comment (m30m) [inline auction-server/src/auction/service/auctionable.rs:64]: Maybe AuctionManager?

File: auction-server/src/kernel/contracts.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            TransactionRequest,
+            H160,
+            U256,
+        },
+    },
+    gas_oracle::EthProviderOracle,
+};
+
+abigen!(
+    OpportunityAdapter,
```
Comment (m30m) [inline auction-server/src/kernel/contracts.rs:36]: why can't we have them in the opportunity module?

File: auction-server/src/kernel/entities.rs [hunk 4]
```diff
@@ -39,3 +52,41 @@ impl<'de> Deserialize<'de> for PermissionKeySvm {
...
+            .filter_map(|log| MulticallIssuedFilter::decode_log(&log.into()).ok())
+            .collect()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Svm;
+
+impl Svm {
+    pub fn check_tx_size(transaction: &VersionedTransaction) -> Result<(), RestError> {
```
Comment (m30m) [inline auction-server/src/kernel/entities.rs:80]: maybe move this inside `verification.rs`

File: auction-server/src/kernel/entities.rs [hunk 5]
```diff
@@ -39,3 +52,41 @@ impl<'de> Deserialize<'de> for PermissionKeySvm {
...
+pub enum ChainType {
+    Evm,
+    Svm,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Evm;
+
+impl Evm {
+    pub fn decode_logs_for_receipt(receipt: &TransactionReceipt) -> Vec<MulticallIssuedFilter> {
```
Comment (m30m) [inline auction-server/src/kernel/entities.rs:66]: let's move this to the same file as the caller. No need to put it inside some impl. Feels very weird here

File: auction-server/src/server.rs [hunk 6]
```diff
@@ -251,91 +244,246 @@ pub async fn start_server(run_options: RunOptions) -> Result<()> {
...
+            )
+        })
+        .collect();
+    chains_svm.iter().for_each(|(chain_id, chain_store)| {
+        if auction_services
+            .insert(
+                chain_id.clone(),
+                auction_service::ServiceEnum::Svm(auction_service::Service::new(
+                    pool.clone(),
+                    auction_service::Config {
```
Comment (m30m) [inline auction-server/src/server.rs:307]: this is such a bit object creation here. Maybe split it into smaller functions?

File: auction-server/src/opportunity/service/mod.rs [hunk 7]
```diff
@@ -48,19 +55,43 @@ mod make_adapter_calldata;
...
     pub weth:                     Address,
+    pub auction_service:          Option<auction_service::Service<Evm>>,
+}
+
+impl ConfigEvm {
+    pub fn inject_auction_service(&self, service: auction_service::Service<Evm>) {
+        // this unsafe block is safe because as this method is called only when setting up services
+        // the method is never called concurrently
+        let selff = self as *const Self as *mut Self;
+        unsafe {
```
Comment (m30m) [inline auction-server/src/opportunity/service/mod.rs:74]: let's discuss this

File: auction-server/src/auction/api.rs [hunk 8]
```diff
@@ -0,0 +1,717 @@
...
+///
+/// This api is deprecated and will be removed soon. Use /v1/{chain_id}/bids/{bid_id} instead.
+#[utoipa::path(get, path = "/v1/bids/{bid_id}",
+    params(("bid_id"=String, description = "Bid id to query for")),
+    responses(
+    (status = 200, body = BidStatus),
+    (status = 400, response = ErrorBodyResponse),
+    (status = 404, description = "Bid was not found", body = ErrorBodyResponse),
+),)]
+#[deprecated = "Use get_bid_status instead"]
```
[the remark this answers is not in the document]
Comment (danimhr) [reply auction-server/src/auction/api.rs:393]: This is not the message that User see :D

File: auction-server/src/kernel/entities.rs (hunk 4 above)
Comment (danimhr) [reply auction-server/src/kernel/entities.rs:80]: Isn't it a general function for Svm chains?

File: auction-server/src/opportunity/service/mod.rs (hunk 7 above)
Comment (danimhr) [reply auction-server/src/opportunity/service/mod.rs:74]: again?

**Review by m30m — APPROVED** [review]

**Pull request merged**